plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java

```diff
@@ -0,0 +1,398 @@
+    public static final String IGNORE_MISSING = "ignore_missing";
+    public static final int DEFAULT_MAX_PREDICTION_TASKS = 10;
+
+    private Configuration suppressExceptionConfiguration = Configuration
+        .builder()
+        .options(Option.SUPPRESS_EXCEPTIONS, Option.DEFAULT_PATH_LEAF_TO_NULL, Option.ALWAYS_RETURN_LIST)
+        .build();
+
+    protected MLInferenceIngestProcessor(
+        String modelId,
+        List<Map<String, String>> inputMaps,
+        List<Map<String, String>> outputMaps,
+        Map<String, String> modelConfigMaps,
+        int maxPredictionTask,
+        String tag,
+        String description,
+        boolean ignoreMissing,
+        boolean ignoreFailure,
+        ScriptService scriptService,
+        Client client
+    ) {
+        super(tag, description);
+        this.inferenceProcessorAttributes = new InferenceProcessorAttributes(
+            modelId,
+            inputMaps,
+            outputMaps,
+            modelConfigMaps,
+            maxPredictionTask
+        );
+        this.ignoreMissing = ignoreMissing;
+        this.ignoreFailure = ignoreFailure;
+        this.scriptService = scriptService;
+        this.client = client;
+    }
+
+    /**
+     * overwrite in this execute method,
+     * when batch inference is available,
+     * to support async multiple predictions.
+     */
+    @Override
+    public void execute(IngestDocument ingestDocument, BiConsumer<IngestDocument, Exception> handler) {
+
+        List<Map<String, String>> processInputMap = inferenceProcessorAttributes.getInputMaps();
+        List<Map<String, String>> processOutputMap = inferenceProcessorAttributes.getOutputMaps();
+        int inputMapSize = (processInputMap != null) ? processInputMap.size() : 0;
+
+        GroupedActionListener<Void> batchPredictionListener = new GroupedActionListener<>(new ActionListener<Collection<Void>>() {
+            @Override
+            public void onResponse(Collection<Void> voids) {
+                handler.accept(ingestDocument, null);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                if (ignoreFailure) {
+                    handler.accept(ingestDocument, null);
+                } else {
+                    handler.accept(null, e);
+                }
+            }
+        }, Math.max(inputMapSize, 1));
+
+        for (int inputMapIndex = 0; inputMapIndex < Math.max(inputMapSize, 1); inputMapIndex++) {
+            try {
+                processPredictions(ingestDocument, batchPredictionListener, processInputMap, processOutputMap, inputMapIndex, inputMapSize);
+            } catch (Exception e) {
+                if (ignoreFailure) {
+                    batchPredictionListener.onFailure(e);
+                } else {
+                    throw e;
+                }
+
+            }
+        }
+    }
+
+    @Override
+    public IngestDocument execute(IngestDocument ingestDocument) throws Exception {
```

Comment:
Let's add comments here. I did not clearly understand why this method should not get executed.

Reply:
added